drivers/pinctrl/pinctrl_ti_k3.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2023 Enphase Energy
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/drivers/pinctrl.h>
+
+int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintptr_t reg)
+{
+	ARG_UNUSED(reg);
+	uintptr_t reg_base = DT_REG_ADDR(DT_NODELABEL(pinctrl));
+
+	for (uint8_t i = 0; i < pin_cnt; i++) {
+		sys_write32(pins[i].value, reg_base + pins[i].offset);
+	}
```

Comment:
This is really something to avoid for MMU-enabled platforms. You are reading and writing a physical address retrieved from the DT implicitly assuming that you have a 1:1 mapping between phys and virt addresses and it is not always the case (as long as you force it).

The MMIO region of the pinctlr driver should map the MMIO space using the device MMIO APIs (see `include/zephyr/sys/device_mmio.h`) and the driver should access the virtual address instead.

Reply:
Fixed, thanks for the feedback!